### ISSUE TITLE
Automatic replies from subscribers are no longer triggered

### ISSUE THREAD

**Issue by CorneliusC** [sympa-community/sympa#1736]
Hello, we recently switched from Sympa 6.2.44 to Sympa 6.2.71b.1. Now we have received information from some users that they no longer receive automatic notifications from subscribers to mailing lists. The feature "reject_mail_from_automates_feature" is set to "off" for the affected lists. The employees would like to know if someone has received the mail in their absence.

After some research, I have identified the following headers that may prevent auto-notifications from being triggered.

Precedence: list => New line in the mails?
Precedence: bulk 
List-Id: <test.domain.abc>

After a brief check, I was only able to determine that the line: Precedence: list is not present in headers of mails that were sent via the old Sympa version. On the other hand, I did not notice any change when I removed the header lines using "remove_headers".

The tests were carried out with both our mail systems (Exchange Online) and external systems (GMX, GoogleMail etc.). 

Has anyone experienced similar problems? According to some entries, Microsoft interprets the line "precedence" somewhat special.

Version
-------
6.2.71b.1

Installation method
-------------------
Source package

Expected behavior
-----------------
Automatic notifications should be delivered.

Actual behavior
---------------
Automatic notifications are not delivered. They are apparently no longer triggered at all.

Additional information
----------------------

**Comment by ikedas** [comment]
Hi @CorneliusC ,

> ... Now we have received information from some users that they no longer receive automatic notifications from subscribers to mailing lists...

What are "automatic notifications from subscribers to mailing lists"? How are they expected to be sent?

Could you please confirm in Sympa's logs that Sympa has indeed received the message in question and has discarded or rejected it?

**Comment by CorneliusC** [comment]
@ikedas Hi ikedas,
this should be the normal behavior, at least it was in the past, if an autoresponder is available, e.g. for vacations:

1. User sends Mail to List > 2. Sympa processes the message > 3. Distribute them to subscribers >
4. Subscriber mail server/client receives the message > 5. Subscriber mail server/client evaluates the headers > 6. Vacation responder is sent to "Reply-To" address from mail-header

As far as we can tell from our logs in the central mail backbone, the automatic notification is no longer triggered at all. Point 6 has therefore been removed, regardless of whether an autoresponder has been activated or not; only points 1-5 are still present in our system.

So Sympa receives the message to be distributed, but the target systems seem to be prevented from sending the automatic message.

If you send a message directly, without a list server, then the automatic notification is triggered as expected.

**Comment by ikedas** [comment]
> As far as we can tell from our logs in the central mail backbone, the automatic notification is no longer triggered at all. Point 6 has therefore been removed, regardless of whether an autoresponder has been activated or not; only points 1-5 are still present in our system.
> 
> So Sympa receives the message to be distributed, but the target systems seem to be prevented from sending the automatic message.
 
It does not necessarily mean that Sympa's behavior has changed. Why do you say that the behavior of the vacation responder has not changed?

> ... 6. Vacation responder is sent to "Reply-To" address from mail-header

If it is the same behavior as before, it means that Sympa used to add a Reply-To header, but does not do so now. It does not necessarily mean that the behavior of Sympa has changed, but it may be that the configuration has been changed by anyone. Check the "Reply address" (reply_to_header) parameter of the list in question.

**Comment by CorneliusC** [comment]
Yes exactly, which is why we have already carried out a number of tests:

"Test mailing list"
- Subscriber of O365 (Exchange Online)
- Subscriber of GMX (German web portal/email service)
- Subscriber of GoogleMail

We activated the auto-notification for each account and sent a message from one account at a time via the "test list". No auto-notification was triggered.

The reply-to notification always points to the sender of the message, i.e. not to the list itself.

When we send directly to each other, the sender receives a notification. 

However, I could not find that Sympa has different values in the headers than in version 2.6.44. And even if I remove all relevant header values from Sympa with "remove_headers", I do not receive an auto-notification.

But if you haven't changed anything here on purpose, then the problem may lie outside. As far as I have read, there are different approaches to the implementation of RFCs, for example at Microsoft. We are just surprised that we also have the same behavior here with external accounts and that this behavior did not occur before the Sympa version change.

**Comment by ikedas** [comment]
If the auto-responses from subscribers are no longer sent to the list because the headers controlled by Sympa have changed, they should be sent to other addresses. Are you observing such messages?

If not, and no auto-responses are being sent at all, then such behavior may be due to the specification of auto-responders.  For example, many "vacation" service implementations will no longer send auto-responses if they receive repeated messages from the same sender within a short period.

**Comment by CorneliusC** [comment]
Yes, testing is a bit time-consuming, I basically have to create a new mailing list for each new test, because otherwise the Exchange server and the extern systems won't trigger a new auto-reply.

I have now explicitly tested the Microsoft Exchange system again. I have noticed that both the external portal and the local Exchange ask for a "read receipts" when the mail is received. If I change the local settings for "Read receipts" in the Exchange account from "Ask me before sending a response" to "Always send a response", then an automatic response is triggered.

However, I don't know why the "read receipts" should now play a role in an out-of-office message. These were three test lists where I did not set up anything like this, not even in the mail accounts, only the auto-response was activated. I think I will do some more internal research here. Thanks for the help! :-)

Translated with www.DeepL.com/Translator (free version)
